sql/core/src/main/scala/org/apache/spark/sql/execution/CacheManager.scala

```diff
@@ -117,12 +117,20 @@ class CacheManager extends Logging with AdaptiveSparkPlanHelper {
       logWarning("Asked to cache already cached data.")
     } else {
       val sessionWithConfigsOff = getOrCloneSessionWithConfigsOff(spark)
+      val completeTableName = tableName.map { table =>
+        import spark.sessionState.analyzer.CatalogAndIdentifier
+        spark.sessionState.sqlParser.parseMultipartIdentifier(table) match {
+          case Seq(_) if spark.catalog.getTable(table).isTemporary => table
+
+          case CatalogAndIdentifier(cat, ident) => s"${cat.name()}.${ident.toString}"
```

Comment:
I think we should ask the caller side to pass in a fully qualified table name, instead of relying on the current catalog/namespace here.

Reply:
makes sense, I will make the changes, thank you!